common/src/main/java/io/netty/util/concurrent/UnorderedThreadPoolEventExecutor.java

```diff
@@ -59,6 +60,7 @@ public UnorderedThreadPoolEventExecutor(int corePoolSize) {
      */
     public UnorderedThreadPoolEventExecutor(int corePoolSize, ThreadFactory threadFactory) {
         super(corePoolSize, threadFactory);
+        this.setThreadFactory(ThreadExecutorMap.apply(getThreadFactory(), this));
```

Comment:
Is this is safe? I think the scheduled repeating trimming task could be run on an arbitrary thread, i.e. not necessarily the one from which it was scheduled that owns the referenced `PoolThreadCache` instance.

Maybe for this one the scheduled task could just call `PoolThreadCache.trimCurrentThreadCache()`?

Reply:
you are right... let me fix it

Reply:
@njhill also with this in mind I wonder if we should maybe restrict all of this to `OrderedEventExecutor`

Reply:
@njhill done that ....

Reply:
Thanks @normanmaurer generally LGTM, but I'm not sure about the `OrderedEventExecutor` check since my understanding is that although `SingleThreadEventExecutor` and `GlobalEventExecutor` are ok, that interface doesn't necessarily imply same-thread semantics, only serialized tasks. So having the check there _might_ be misleading... it's necessary but not sufficient. Maybe we could have a static `scheduledTasksRunOnSameThread(EventExecutor)` helper method which for now returns true only in the case of STEE and GEE?

(but I still like the change to make GEE implement OEE!)

Reply:
@normanmaurer that's probably fine but technically there's still a chance that someone has subclassed it with different `scheduleAtFixedRate` behaviour that breaks the assertion, since the method is non-final (though this wouldn't really make much sense to do).

Reply:
@njhill true.. I guess we could also just return directly if the `Thread` is not the same we expect. WDYT ?

Reply:
@njhill alternatively I just noticed we could just call:

```java
PooledByteBufAllocator.this.trimCurrentThreadCache();
```

This should be safe in any case.

Reply:
@normanmaurer yes, that would be safe for any of the cases (worst case it would be a repeating no-op)... that's actually what I was meaning to suggest [above](https://github.com/netty/netty/pull/8941#discussion_r266107986) but obviously got the detail wrong since that method is on the BBAllocator not PoolThreadCache!